clouddriver-openstack/src/test/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/CloneOpenstackAtomicOperationSpec.groovy

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2016 Veritas Technologies LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.clouddriver.openstack.deploy.ops.servergroup
+
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.openstack.client.OpenstackClientProvider
+import com.netflix.spinnaker.clouddriver.openstack.client.OpenstackProviderFactory
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.CloneOpenstackAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.DeployOpenstackAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackCredentials
+import com.netflix.spinnaker.clouddriver.openstack.security.OpenstackNamedAccountCredentials
+import org.openstack4j.openstack.heat.domain.HeatStack
+import spock.lang.Specification
+import spock.lang.Subject
+
+class CloneOpenstackAtomicOperationSpec extends Specification {
+  private static final String ACCOUNT_NAME = 'myaccount'
+  private static final APPLICATION = "app"
+  private static final STACK = "stack"
+  private static final DETAILS = "details"
+  private static final REGION = "region"
+  private static final String HEAT_TEMPLATE = "{\"heat_template_version\":\"2013-05-23\"," +
+    "\"description\":\"Simple template to test heat commands\"," +
+    "\"parameters\":{\"flavor\":{\"default\":\"m1.nano\",\"type\":\"string\"}}," +
+    "\"resources\":{\"hello_world\":{\"type\":\"OS::Nova::Server\"," +
+    "\"properties\":{\"flavor\":{\"get_param\":\"flavor\"}," +
+    "\"image\":\"cirros-0.3.4-x86_64-uec\",\"user_data\":\"\"}}}}"
+  private static final Integer TIMEOUT_MINS = 5
+  private static final Map<String,String> PARAMS_MAP = Collections.emptyMap()
+  private static final Boolean DISABLE_ROLLBACK = false
+
+  private static final SEQUENCE = "v000"
+  private static final ANCESTOR_STACK_NAME = "$APPLICATION-$STACK-$DETAILS-$SEQUENCE"
+  private static final ANCESTOR_STACK_ID = "978f378f-0b98-469d-90dd-f61a73f8703a"
+
+  // Changed Parameters
+  private static final STACK_N = "stackn"
+  private static final APPLICATION_N = "appn"
+  private static final DETAILS_N = "detailn"
+  private static final REGION_N = "regionn"
+  private static final String HEAT_TEMPLATE_N = "{\"heat_template_version\":\"2013-05-24\"," +
```

Comment:
If you use single quotes for the entire template, you shouldn't need to escape all the double quotes in the JSON.